src/main/java/org/spongepowered/api/ai/goal/GoalSelector.java

```diff
@@ -58,24 +59,24 @@
     O getOwner();
 
     /**
-     * Adds a new {@link org.spongepowered.api.ai.goal.Goal} to this {@link GoalSelector}.
+     * Adds a new {@link Goal} to this {@link GoalSelector}.
      *
      * @param priority The priority this goal should run at
      * @param task The goal to run
      * @return This goal, for chaining
      */
-    GoalSelector addTask(int priority, org.spongepowered.api.ai.goal.Goal<O> task);
+    GoalSelector<O> addTask(int priority, Goal<O> task);
```

Comment:
Maybe use `Goal<? super O>`. That way you could reuse a goal builder for different monster types e.g. zombies and skeletons.

Reply:
Nah. Now the `O` is the exact type, no longer fuzzy like before.